plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergUtil.java

```diff
@@ -397,4 +398,19 @@ public static Transaction newCreateTableTransaction(TrinoCatalog catalog, Connec
 
         return catalog.newCreateTableTransaction(session, schemaTableName, schema, partitionSpec, targetPath, propertiesBuilder.build());
     }
+
+    public static long getSnapshotIdAsOfTime(Table table, long timestampMillis)
+    {
+        Long snapshotId = null;
+        for (HistoryEntry logEntry : table.history()) {
+            if (logEntry.timestampMillis() <= timestampMillis) {
+                snapshotId = logEntry.snapshotId();
+            }
+        }
+
+        if (snapshotId == null) {
+            throw new TrinoException(GENERIC_USER_ERROR, "Cannot find a snapshot older than epoch time " + timestampMillis);
```

Comment:
From a user perspective, if the oldest history timestamp is shown, I think it should be formatted to be easily readable.

Use for reference (in case you want to show the timestamp to the user) :
-  https://github.com/trinodb/trino/blob/master/plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/HistoryTable.java#L85 
-  https://github.com/trinodb/trino/blob/master/core/trino-spi/src/main/java/io/trino/spi/type/Timestamps.java#L147 (this corresponds to the what is being seen in the `trino cli` when retrieving `select current_timestamp;`)

Reply:
Thanks, I have made the changes (similar to the Snowflake's message).